qubes/ext/core_features.py

```diff
@@ -62,3 +62,23 @@ def qubes_features_request(self, vm, event, untrusted_features):
             # if this is the first time qrexec was advertised, now can finish
             #  template setup
             yield from vm.fire_event_async('template-postinstall')
+
+    # pylint: disable=no-self-use
+    def set_servicevm_feature(self, subject):
+        if subject.provides_network:
+            subject.features['servicevm'] = 1
+        elif 'servicevm' in subject.features:
+            del subject.features['servicevm']
+
+    @qubes.ext.handler('property-set:provides_network')
+    def on_property_set(self, subject, _event, _name, _newvalue,
```

Comment:
This won't work - event's arguments are keyword arguments so must have specific names (see tests that fail now).